python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+
+        The dictionary should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct a calibration from a dictionary of properties.
+
+        Must be implemented by the specific calibration subclasses.
+
+        Parameters
+        ----------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables of properties.
+
+        Returns
+        ------
+        calib : `lsst.ip.isr.CalibType`
+            Constructed calibration.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    def toTable(self):
+        """Return a list of tables containing the calibration properties.
+
+        The table list should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables of properties.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    def validate(self, other=None):
+        """Validate that this calibration is defined and can be used.
+
+        Parameters
+        ----------
+        other : object, optional
+            Thing to validate against.
+
+        Returns
+        -------
+        valid : `bool`
+            Returns true if the calibration is valid and appropriate.
+        """
+        return False
+
+    def apply(self, target):
+        """Method to apply the calibration to the target object.
+
+        Parameters
+        ----------
+        target : object
+            Thing to validate against.
+
+        Returns
+        -------
+        valid : `bool`
+            Returns true if the calibration was applied correctly.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+
+class IsrProvenance(IsrCalib):
+    """Class for the provenance of data used to construct calibration.
+
+    Provenance is not really a calibration, but as we would like to
+    record this when constructing the calibration, and it provides an
+    example of the base calibration class.
+
+    Parameters
+    ----------
+    registryName : `str`, optional
+        Name of the butler registry the dataIds were obtained from.
+    calibType : `str`, optional
+        Type of calibration this provenance was generated for.
```

Comment:
Is this the gen3 dataset type name?

Reply:
My thought was OtherCalib._OBSTYPE, but I would expect that to be identical to the gen3 dataset_type_name.